tutorial/source/tensor_shapes.ipynb

```diff
@@ -0,0 +1,292 @@
+   "metadata": {},
+   "source": [
+    "Another way to batch distributions is via the `.reshape()` method. This only works if \n",
+    "parameters are identical along the leftmost dimensions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "d = Bernoulli(torch.tensor([0.1, 0.2, 0.3, 0.4])).reshape([3])\n",
+    "assert d.batch_shape == (3, 4)\n",
+    "assert d.event_shape == ()\n",
+    "x = d.sample()\n",
+    "assert x.shape == (3, 4)\n",
+    "assert d.log_prob(x).shape == (3, 4)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Multivariate distributions have nonempty `.event_shape`. For these distributions, the shapes of `.sample()` and `.log_prob(x)` differ:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "d = MultivariateNormal(torch.zeros(3), torch.eye(3, 3))\n",
+    "assert d.batch_shape == ()\n",
+    "assert d.event_shape == (3,)\n",
+    "x = d.sample()\n",
+    "assert x.shape == (3,)            # == batch_shape + event_shape\n",
+    "assert d.log_prob(x).shape == ()  # == batch_shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In Pyro you can treat a univariate distribution as multivariate by calling the `.reshape(extra_event_dims=_)` property."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "d = Bernoulli(0.5 * torch.ones(3,4)).reshape(extra_event_dims=1)\n",
+    "assert d.batch_shape == (3,)\n",
+    "assert d.event_shape == (4,)\n",
+    "x = d.sample()\n",
+    "assert x.shape == (3, 4)\n",
+    "assert d.log_prob(x).shape == (3,)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "While you work with Pyro programs, keep in mind that samples have shape `batch_shape + event_shape`, whereas `.log_prob(x)` values have shape `batch_shape`. You'll need to ensure that `batch_shape` is carefully controlled by either trimming it down with `.reshape(extra_event_dims=_)` or by declaring dimensions as independent via `pyro.iarange`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Declare independent dims with `iarange`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def model():\n",
+    "    a = sample(\"a\", Normal(0, 1))\n",
+    "    b = sample(\"b\", Normal(zeros(2), 1).reshape(extra_event_dims=1))\n",
+    "    with iarange(\"c\", 2):\n",
+    "        c = sample(\"c\", Normal(zeros(2), 1))\n",
+    "    with iarange(\"d\", 3):\n",
+    "        d = sample(\"d\", Normal(zeros(3,4,5), 1).reshape(extra_event_dims=2))\n",
+    "    assert a.shape == ()\n",
+    "    assert b.shape == (2,)\n",
+    "    assert c.shape == (2,)\n",
```

Comment:
add some comments for what these shapes are and their independence along what dimensions